modules/xfem/include/base/XFEM.h

```diff
@@ -417,33 +474,59 @@ class XFEM : public XFEMInterface
    * @param parent_elem The parent element
    * @param elem1       The first child element
    * @param elem2       The second child element
    */
-  void storeMaterialPropertiesForElements(const Elem * parent_elem,
-                                          const std::vector<const Elem *> & elems,
-                                          const GeometricCutUserObject * gcuo);
+  void storeMaterialPropertiesForElement(const Elem * parent_elem, const Elem * child_elem);
+
+  /**
+   * Load the material properties
+   * @param props_deserialized The material properties
+   * @param props_serialized   The serialized material properties
+   */
+  void loadMaterialProperties(
+      HashMap<unsigned int, MaterialProperties> props_deserialized,
+      const std::unordered_map<unsigned int, std::string> & props_serialized) const;
+
+  /**
+   * Load the material properties
+   * @param props_deserialized The material properties
+   * @param props_serialized   The serialized material properties
+   */
+  void loadMaterialPropertiesForElementHelper(const Elem * elem,
+                                              const Xfem::CachedMaterialProperties & cached_props,
+                                              const MaterialPropertyStorage & storage) const;
 
   /**
    * Helper function to store the material properties of a healed element
-   * @param parent_elem The parent element
-   * @param cut_elem    The element being cut
-   * @param elem_from   The element to copy material properties from
+   * @param elem         The cut element to restore material properties to.
+   * @param elem_from    The element to copy material properties from.
+   * @param cached_gcei  The material properties cache to use.
    */
-  void setMaterialPropertiesForElement(const Elem * parent_elem,
-                                       const Elem * cut_elem,
-                                       const GeometricCutUserObject * gcuo);
+  void loadMaterialPropertiesForElement(
+      const Elem * elem,
+      const Elem * elem_from,
+      std::unordered_map<const Elem *, Xfem::GeometricCutElemInfo> & cached_gcei) const;
 
   /**
-   * Determine which cut subdomain of the element belongs to relative to the cut
-   * @param parent_elem The parent element
-   * @param cut_elem    The element being cut
-   * @param gcuo        The GeometricCutUserObject for the cut
+   * Return the first node in the provided element that is found to be in the physical domain
+   * @param e  Constant pointer to the child element
+   * @param e0 Constant pointer to the parent element whose nodes will be querried
+   * @return A constant pointer to the node
    */
-  GeometricCutSubdomainID getGeometricCutSubdomainID(const Elem * parent_elem,
-                                                     const Elem * cut_elem,
-                                                     const GeometricCutUserObject * gcuo) const;
+  const Node * pickOnePhysicalNode(const Elem * e, const Elem * e0) const;
```

Comment:
It might be clearer to rename this to `pickFirstPhysicalNode`, but do as you see fit.